docs: document ssrWidth to prevent mobile hydration errors

mathieumagalhaes opened the pull request:
<!---☝️ PR title should follow conventional commits (https://conventionalcommits.org) -->

### 🔗 Linked issue
https://github.com/unovue/shadcn-vue/issues/1366

<!-- Please ensure there is an open issue and mention its number as #123 -->

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->
When using any component that relies on useMediaQuery in a Nuxt SSR setup,
a hydration error occurs due to the absence of an ssrWidth value.
https://github.com/unovue/shadcn-vue/issues/1366

**Solution:**
According to the [VueUse documentation](https://vueuse.org/core/useMediaQuery/#server-side-rendering-and-nuxt), the best practice is to provide an ssrWidth to prevent this error. By doing so, we can ensure the component works correctly without affecting other users or breaking existing functionality.

I also ran `lint:fix`


### 📸 Screenshots (if appropriate)

<!-- Add screenshots to help explain the change. -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

mathieumagalhaes committed:
fix(Sidebar): set a ssrWidth on useMediaQuery

sadeghbarati reviewed (commented):
Hi 👋 

Thanks for the PR, 

Looking to vueuse [`useMediaQuery`](https://github.com/vueuse/vueuse/blob/main/packages/core/useMediaQuery/index.ts#L20) source,
How about adding document about the [`useSSRWidth`](https://vueuse.org/core/useSSRWidth/) so we could prevent duplicated `ssrWidth` option in the code?

for example adding documents here https://www.shadcn-vue.com/docs/installation/nuxt

mathieumagalhaes committed:
Revert "fix(Sidebar): set a ssrWidth on useMediaQuery"

This reverts commit 11b6b1bba27a91f073f792f5769aa9eff4655464.

mathieumagalhaes committed:
docs: add documentation around ssrWidth to prevent mobile hydration errors

mathieumagalhaes commented:
> Hi 👋
> 
> Thanks for the PR,
> 
> Looking to vueuse [`useMediaQuery`](https://github.com/vueuse/vueuse/blob/main/packages/core/useMediaQuery/index.ts#L20) source, How about adding document about the [`useSSRWidth`](https://vueuse.org/core/useSSRWidth/) so we could prevent duplicated `ssrWidth` option in the code?
> 
> for example adding documents here https://www.shadcn-vue.com/docs/installation/nuxt

Hi @sadeghbarati 
Great catch, you're absolutely right. Using provideSSRWidth is a cleaner and more scalable solution moving forward.
I've reverted my previous changes and added the appropriate documentation.

mathieumagalhaes commented on packages/cli/src/utils/get-config.ts:
```diff
@@ -4,9 +4,9 @@ import path from 'pathe'
 import { glob } from 'tinyglobby'
 import { z } from 'zod'
 import { getProjectInfo } from '@/src/utils/get-project-info'
+import { logger } from '@/src/utils/logger'
```
PS. This has been moved by running lint:fix

sadeghbarati committed:
Merge branch 'dev' of https://github.com/unovue/shadcn-vue into dev

sadeghbarati committed:
chore: update

All plugins inside are auto-registered, you don't need to add them to your nuxt.config separately.

sadeghbarati committed:
chore: add link to vueuse useSSRWidth
